packages/common/src/messages/responses/generate-api-token.ts

```diff
@@ -0,0 +1,61 @@
+import {ResponseBase, ResponseError, ResponseSuccess} from './response-base';
+import {_GenerateApiTokenResponse} from './grpc-response-types';
+import {SdkError} from '../../errors';
+
+export abstract class Response extends ResponseBase {}
+
+class _Success extends Response {
+  readonly apiToken: string;
+  readonly refreshToken: string;
+  readonly endpoint: string;
+  readonly validUntil: number;
+
+  constructor(result?: _GenerateApiTokenResponse) {
```

Comment:
should this be required? If this is optional, then your getters will return optional strings